src/main/java/org/apache/maven/plugins/dependency/filters/ArtifactIdFilter.java

```diff
@@ -0,0 +1,98 @@
+package org.apache.maven.plugins.dependency.filters;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.util.HashSet;
+import java.util.Set;
+
+import org.apache.maven.model.Dependency;
+
+/**
+ * Filters dependencies by artifact ID.
+ */
+public class ArtifactIdFilter extends AbstractDependencyFilter
+{
+    private final String includeArtifactId;
+
+    private final String excludeArtifactId;
+
+    public ArtifactIdFilter( String includeArtifactId, String excludeArtifactId )
```

Comment:
are these plural? since they have to be split? That is are these really includeArtifactIds and excludeArtifactIds? If so, it might make sense to pass a collection of each instead. 

What happens if the same artifact ID is in both lists?

Group IDs not considered here? 


In either case it could use a doc comment.

Reply:
Another copy/paste from other parts of the mojo. We have the choice of consistency vs doing it right.

Those come from the mojo as string parameter, thus can't change it. Have to split it here. But will change to plural.

Reply:
The documentation does not state what should happen if it is in both lists. In this case it will be excluded. That should be fine and is not surprising behaviour.